nextcord/user.py

```diff
@@ -85,6 +87,74 @@ def asset(self) -> Asset:
         return Asset._from_avatar_decoration(self.user._state, self._asset)
 
 
+class PrimaryGuild:
+    """Represents a primary guild for a user.
+
+    .. versionadded:: 3.2
+
+    Attributes
+    ----------
+    identity_guild_id: Optional[:class:`int`]
+        The ID of the primary guild.
+    identity_enabled: Optional[:class:`bool`]
+        Whether displaying the primary guild is enabled.
+
+        This is ``None`` if the user has not accepted a change to the tag details.
+    tag: Optional[:class:`str`]
+        The tag of the primary guild.
+    badge: Optional[:class:`str`]
+        The badge of the primary guild.
+    """
+
+    __slots__ = (
+        "_state",
+        "identity_guild_id",
+        "identity_enabled",
+        "tag",
+        "_badge",
+    )
+
+    def __init__(self, *, data: UserPrimaryGuildPayload, state: ConnectionState) -> None:
+        self._state: ConnectionState = state
+        self.identity_guild_id: Optional[int] = get_as_snowflake(data, "identity_guild_id")
+        self.identity_enabled: Optional[bool] = data.get("identity_enabled")
+        self.tag: Optional[str] = data.get("tag")
+        self._badge: Optional[str] = data.get("badge")
+
+    @classmethod
+    def _empty(cls, state: ConnectionState) -> Self:
+        """Creates an empty :class:`PrimaryGuild`.
+
+        This is a better alternative to returning ``None`` when the primary guild is not set,
+        as all of these fields are nullable anyway, allows for easier checking of the primary guild.
+        """
+        return cls(
+            data={
+                "identity_guild_id": None,
+                "identity_enabled": False,
+                "tag": None,
+                "badge": None,
+            },
+            state=state,
+        )
+
+    @property
+    def badge(self) -> Optional[Asset]:
+        """Optional[:class:`Asset`]: Returns an :class:`Asset` for the badge of the primary guild."""
+        if self._badge is not None and self.identity_guild_id is not None:
+            return Asset._from_primary_guild_badge(
+                self._state, guild_id=self.identity_guild_id, badge_hash=self._badge
+            )
+        return None
+
```

Comment:
You can get the creation date of the primary guild
```suggestion

    @property 
    def created_at(self) -> Optional[datetime]:
        """Optional[:class:`datetime.datetime`]: Returns the primary guild's creation time in UTC."""
        if self.identity_guild_id:
            return snowflake_time(self.identity_guild_id)
        return None

```